src/test/rustdoc/issue-74134-2.rs

```diff
@@ -0,0 +1,11 @@
+// compile-flags: --document-private-items
+#![deny(intra_doc_link_resolution_failure)]
```

Comment:
I think this can be condensed into one test case, see https://rustc-dev-guide.rust-lang.org/tests/adding.html?highlight=ui#revisions. That way we don't have to duplicate the whole file just to pass `--document-private-items`.

Reply:
All tests are now squashed into a single file and also expanded to cover running with and without `--document-private-items`.

I had to move them over to `rustdoc-ui` to be able to use revisions, because regular rustdoc tests aren't allowed to ([runtest.rs#L2293]). Thus, it now tests rustdoc's output as well.

[runtest.rs#L2293]: https://github.com/rust-lang/rust/blob/daecab3a784f28082df90cebb204998051f3557d/src/tools/compiletest/src/runtest.rs#L2293

Reply:
Looks great! Thanks for the work on this :)